src/entities/Basket/Basket.controller.ts

```diff
@@ -0,0 +1,27 @@
+import StatusCodes from 'http-status-codes';
+import { Request, Response } from 'express';
+// import validateBasketReq from './Basket.validation';
+import Basket from './Basket.schema';
+
+const { BAD_REQUEST, CREATED, OK } = StatusCodes;
+
+export const updateBasket = async (req: Request, res: Response) => {
+    
+    const basketStatus = req.body;
+    try {
+        const basket = await Basket
+        .findOneAndUpdate(
+        { _id: req.params.id }, 
+        { $set: basketStatus }, 
+        { new: true })
+        .populate('createdByUserId', 'name')
+        .populate('targetUserID', 'name')
+        
+        if (!basket) {
+            return res.send(BAD_REQUEST).send('There is no basket to be updated')
+        }
+        return res.status(OK).json(basket);
+    } catch (error) {
+        return res.status(BAD_REQUEST).send(error.message);
+    }
+}
```

Comment:
Please add validation to make sure that only the user who takes part in the transaction can change its status (use JWT). 

+there could be some complicated logic, for example you shouldn't be able to change the basket status to offered once it's pending ect. but I don't know if it is necessary